### ISSUE TITLE
Deep nested block should emit issues only on outer block when embedded blocks are also too nested

### ISSUE THREAD

**Issue by thelvis4** [oclint/oclint#259]
`Deep nested block` rule reports two violations for the same Obj-C method. One time for the entire method, and second time for an internal nested block.

For the following method:

``` Obj-C
51 - (void)updateSearchDictionaryFromFacets:(NMListFacets *)listFacets {
52      NSArray *validFacets = [listFacets getValidFacetKeys];
53 
54     for (NSString *facetKey in validFacets) {
55          NMSFacetGroup *facetObject = [listFacets getDataForFacetKey:facetKey];
56          NSArray *facetsData = [facetObject getSelectedFacetsString];
57          if (facetsData) {
58              self.searchDict[facetKey] = facetsData;
59          } else {
60              if (![facetKey isEqualToString:@"category"]) {
61                  [self.searchDict removeObjectForKey:facetKey];
62              } else {
63                 if (self.defaultCategory) {
64                      self.searchDict[facetKey] = self.defaultCategory;
65                  }
66              }
67          }
68      }
69  }
```

it  reports:

``` xml
<file name="/Full/Path/To/File.m">
<violation begincolumn="69" endcolumn="1" beginline="51" endline="69" priority="5" rule="deep nested block">
Block depth of 5 exceeds limit of 3
</violation>
</file>

<file name="/Full/Path/To/File.m">
<violation begincolumn="45" endcolumn="5" beginline="54" endline="68" priority="5" rule="deep nested block">
Block depth of 4 exceeds limit of 3
</violation>
</file>
```

**Comment by ryuichi-assistant** [comment]
There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved with the latest updates.

Please make sure to update to the latest version and check if that solves the issue. Let us know if that works for you by adding a comment.

**Comment by ryuichi-assistant** [comment]
This issue will be auto-closed because there hasn't been any activity for a few months. Feel free to open a new one if you still experience this problem.
